### PR TITLE
refactor: create generic progress bar

### PR DESCRIPTION
**Pull request by itsmdasifraza**
## Description
create generic progress bar
Fixes issue #141 

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

### CODE REVIEW

File: src/lib/components/Transition/progress-bar/ProgressBar.svelte [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+          aria-valuemax="100"
+        ></div>
+      {/if}
+    </div>
+  </div>
+</div>
+
+<style>
+  .sparrow-progress-bg-overlay {
+    position: fixed;
```
Comment (parthdalaltech) [inline src/lib/components/Transition/progress-bar/ProgressBar.svelte:49]: class p-fixed check from bootstrap

File: src/lib/components/Transition/progress-bar/ProgressBar.svelte [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    top: 0;
+    left: 0;
+    width: 100vw;
+    height: 100vh;
+    background: var(--background-hover);
+    backdrop-filter: blur(3px);
+  }
+  .progress {
+    position: fixed;
```
Comment (parthdalaltech) [inline src/lib/components/Transition/progress-bar/ProgressBar.svelte:58]: check class from boostrap

File: src/lib/components/Transition/progress-bar/ProgressBar.svelte [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    transform: translate(-50%, -50%);
+    background-color: transparent !important;
+  }
+  .progresser {
+    background-color: var(--send-button);
+    height: 10px;
+  }
+
+  .container {
+    flex-direction: column;
```
Comment (parthdalaltech) [inline src/lib/components/Transition/progress-bar/ProgressBar.svelte:72]: fd-c class from bootstrap

File: src/lib/components/Transition/progress-bar/ProgressBar.svelte [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    background-color: transparent !important;
+  }
+
+  .syntax-highlighting {
+    line-height: 18px;
```
Comment (parthdalaltech) [inline src/lib/components/Transition/progress-bar/ProgressBar.svelte:77]: check. bootstrap class

**Review by parthdalaltech — CHANGES_REQUESTED** [review]

File: src/lib/components/Transition/progress-bar/ProgressBar.svelte (hunk 1 above)
Comment (itsmdasifraza) [reply src/lib/components/Transition/progress-bar/ProgressBar.svelte:49]: resolved

File: src/lib/components/Transition/progress-bar/ProgressBar.svelte (hunk 2 above)
Comment (itsmdasifraza) [reply src/lib/components/Transition/progress-bar/ProgressBar.svelte:58]: resolved

File: src/lib/components/Transition/progress-bar/ProgressBar.svelte (hunk 3 above)
Comment (itsmdasifraza) [reply src/lib/components/Transition/progress-bar/ProgressBar.svelte:72]: resolved

File: src/lib/components/Transition/progress-bar/ProgressBar.svelte (hunk 4 above)
Comment (itsmdasifraza) [reply src/lib/components/Transition/progress-bar/ProgressBar.svelte:77]: resolved

**Review by parthdalaltech — APPROVED** [review]

**Comment by parthdalaltech** [comment]
approved

**Pull request merged**